demo/angular-editable-text.js

```diff
@@ -42,49 +43,80 @@
 
           elem.addClass('gg-editable-text');
 
-          scope.$watch('isEditing', function (val, oldVal) {
+          scope.onInputClick = function() {
+            scope.isEditing = true;
+            if (attrs.hasOwnProperty('selectAll')) {
+              input.select();
+            }
+          };
+
+          $timeout(function() {
```

Comment:
This one actually fixes something else. We want the input width to fill it's container, even if the text doesn't. We resize it after the DOM has loaded, to ensure anywhere within the visible borders is clickable.